arbor/include/arbor/morph/primitives.hpp

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <algorithm>
+#include <cstdlib>
+#include <ostream>
+#include <vector>
+
+#include <arbor/util/lexcmp_def.hpp>
+
+//
+//  Types used to identify concrete locations.
+//
+
+namespace arb {
+
+using msize_t = std::size_t;
+constexpr msize_t mnpos = msize_t(-1);
+
+// a morphology sample point: a 3D location and radius.
+struct mpoint {
+    double x, y, z;  // [µm]
+    double radius;   // [μm]
+
+    friend std::ostream& operator<<(std::ostream&, const mpoint&);
+};
+
+mpoint lerp(const mpoint& a, const mpoint& b, double u);
+bool is_collocated(const mpoint& a, const mpoint& b);
+double distance(const mpoint& a, const mpoint& b);
+
+
+// A morphology sample consists of a location and an integer tag.
+// When loaded from an SWC file, the tag will correspond to the SWC label,
+// which are standardised as follows:
+//  1 - soma
+//  2 - axon
+//  3 - (basal) dendrite
+//  4 - apical dendrite
+// http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html
+// However, any positive integer tag can be provided and labelled dynamically.
+
+struct msample {
+    mpoint loc;
+    int tag;
+
+    friend std::ostream& operator<<(std::ostream&, const msample&);
+};
+
+bool is_collocated(const msample& a, const msample& b);
+double distance(const msample& a, const msample& b);
+
+using point_prop = std::uint8_t;
+constexpr point_prop point_prop_mask_none = 0;
+constexpr point_prop point_prop_mask_root = 1;
+constexpr point_prop point_prop_mask_fork = 2;
+constexpr point_prop point_prop_mask_terminal = 4;
+constexpr point_prop point_prop_mask_collocated = 8;
+
+std::ostream& operator<<(std::ostream& o, const point_prop& p);
```

Comment:
Because `point_prop` is a type alias for `uint8_t`, this is problematic. I don't believe it will ever get found in preference to the `std` namespace overload for `unsigned char`.

Reply:
I will remove. It was just for debugging, and it worked for me.

Reply:
It worked sometimes: certainly when using inside functions in the `arb::` namespace while debugging.